### ISSUE TITLE
funced not returning shell due to $EDITOR Variable

### ISSUE THREAD

**Issue by tuxvador** [fish-shell/fish-shell#6975]
**Vesion** :  3.1.1
**OS** : Manjaro
**Terminal** : terminator
**Uname** : Linux tuxvador-pentest 5.6.8-1-MANJARO #1 SMP PREEMPT Wed Apr 29 19:05:17 UTC 2020 x86_64 GNU/Linux
Encrypted Install
**Term** : screen-256color

Hi devs, first i wanted to say i love what fish is, it helps me being more effective in my everyday tasks.
I encountered an issue with fish today, i wasn't able to call the funced function i kept being stuck without returning back a shell to me. After some investigation, i determined that the issue was due to the editor variable. I haven't yet determined the exact cause of the error but i have determined it was caused between the line 97 and 107. 

I added ans asciinema video showing the error i encontered. 

94
```fish
    while true
        set -l checksum (__funced_md5 "$tmpname")

        if not eval $editor $tmpname
            echo (_ "Editing failed or was cancelled")
        else
            # Verify the checksum (if present) to detect potential problems
            # with the editor command
            if set -q checksum[1]
                set -l new_checksum (__funced_md5 "$tmpname")
                if test "$new_checksum" = "$checksum"
                    echo (_ "Editor exited but the function was not modified")
                end
            end
```
107

asciinema : [https://asciinema.org/a/CrQIIeJFY9lJoymsokwO9V44d](https://asciinema.org/a/CrQIIeJFY9lJoymsokwO9V44d)

**Comment by krobelus** [comment]
> Vim : Alerte : La sortie ne s'effectue pas sur un terminal

Vim prints this warning because stdout is not a terminal. Vim is called in `eval $editor $tmpname`, because fish 3.1.1. buffers `eval` output too aggressively. It is fixed in 3.1.2, which should hit the repositories soon.

Incidentally, the eval here is actually avoidable: we've changed some similar occurrences to use 
`echo $EDITOR | read -at editor` to split the editor, and then simply call `$editor $tmpname`.

**Comment by faho** [comment]
Duplicate of #6955
